src/v/wasm/logger.h

```diff
@@ -16,4 +16,34 @@
 namespace wasm {
 // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
 extern ss::logger wasm_log;
+
+class logger {
+public:
+    struct meta {
+        ss::log_level level;
+        std::string_view transform_name;
+    };
+
+    logger() = delete;
+    explicit logger(ss::sstring name, ss::logger* log)
+      : _name(std::move(name))
+      , _log(log) {}
+    virtual ~logger() = default;
+    logger(const logger&) = delete;
+    logger& operator=(const logger&) = delete;
+    logger(logger&&) = delete;
+    logger& operator=(logger&&) = delete;
+
+    virtual void log(ss::log_level lvl, std::string_view message) noexcept {
```

Comment:
There is no need to make this `virtual` yet right? If we're going to expose this type for other subsystems to implement, I generally prefer a pure virtual abstract class as the interface (for example the transform logger won't need the `ss::logger` field when we have the log manager).

This will mean you'll need two implementations, one in the transform subsystem that for now will write to the transform log, and another in the wasm testing layer that can either noop or write directly to the log.

Reply:
eh, it's a good point. let me refactor this into `api.h` and it should look tidier.